### PR TITLE
Retry and timeout

### PR DESCRIPTION
**Pull request by thomasrockhu**
## Purpose
Push the retries to the `curl` command. This fixes issues with calling for the `s3` too many times.
Add `connect-timeout` which should help timeouts fail quickly.

## Notable Changes
Tested on linux and mac, need to test on Windows

## Tests and Risks?
Windows...

## Update the SHA hash files
Done

### CODE REVIEW

File: codecov [hunk 1]
```diff
@@ -1666,83 +1666,66 @@ else
...
-        say "${s3target}"
+    say "${e}->${x}  Pinging Codecov"
+    say "$url/upload/v4?$queryNoToken"
+    # shellcheck disable=SC2086,2090
+    res=$(curl $curl_s -X POST $curlargs $cacert \
+          --retry 5 --retry-delay 2 --connect-timeout 2 \
+          -H 'X-Reduced-Redundancy: false' \
+          -H 'X-Content-Type: application/x-gzip' \
+          "$url/upload/v4?$query" || true)
+    # a good replay is "https://codecov.io" + "\n" + "https://codecov.s3.amazonaws.com/..."
```
Comment (hootener) [inline codecov:1677]: while you're here, can we fix this typo? `replay` == `reply`
Additionally, the response is no longer `codecov.s3.amazonaws.com`, we should change this to the GCS domain (or remove the comment altogether)

**Comment by thomasrockhu** [comment]
Apparently I dismissed your review @hootener

**Comment by thomasrockhu** [comment]
Grrr, merging master marks reviews as stale

File: SHA1SUM [hunk 2]
```diff
@@ -1 +1 @@
-7f1666285879b50c0c6095eb22b88e3fee1b83d6  codecov
+9e28da66cba4ff80091a7a7d15364517b1cc3391  codecov
```
Comment (thomasrockhu) [inline SHA1SUM:1]:
```suggestion
9e28da66cba4ff80091a7a7d15364517b1cc3391  codecov
```

**Pull request merged**